components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/ApiResourceManagementDAOImpl.java

```diff
@@ -0,0 +1,875 @@
+
+    @Override
+    public List<ApiResource> getApiResources(Integer limit, Integer tenantId, String sortOrder,
+                                             List<ExpressionNode> expressionNodes)
+            throws ApiResourceManagementException {
+
+        return getApiResourcesList(limit, tenantId, sortOrder, expressionNodes);
+    }
+
+    @Override
+    public Integer getApiResourcesCount(Integer tenantId, List<ExpressionNode> expressionNodes)
+            throws ApiResourceManagementException {
+
+        // Remove the after and before attributes from the expression nodes list,
+        // since we are not considering them when getting the total count.
+        List<ExpressionNode> expressionNodesCopy = new ArrayList<>(expressionNodes);
+        if (!expressionNodes.isEmpty()) {
+            expressionNodesCopy.removeIf(expressionNode -> AFTER.equals(expressionNode.getAttributeValue()) ||
+                    BEFORE.equals(expressionNode.getAttributeValue()));
+        }
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNodesCopy, filterQueryBuilder, false);
+
+        Map<Integer, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        String sqlStmt;
+        String getApiResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL;
+        if (ApiResourceManagementUtil.isOracleDB()) {
+            getApiResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL_ORACLE;
+        } else if (ApiResourceManagementUtil.isMSSqlDB()) {
+            getApiResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL_MSSQL;
+        } else if (ApiResourceManagementUtil.isDB2()) {
+            getApiResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL_DB2;
+        } else if (ApiResourceManagementUtil.isH2() || ApiResourceManagementUtil.isMySQL()
+                || ApiResourceManagementUtil.isMariaDB()) {
+            getApiResourcesCountSqlStmtTail = SQLConstants.GET_API_RESOURCES_COUNT_TAIL_H2;
+        }
+
+        sqlStmt = SQLConstants.GET_API_RESOURCES_COUNT + filterQueryBuilder.getFilterQuery() +
+                getApiResourcesCountSqlStmtTail;
+
+        Connection dbConnection = getConnection(false);
+        try {
+            PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt);
+            for (Map.Entry<Integer, String> prepareStatement : filterAttributeValue.entrySet()) {
+                prepStmt.setString(prepareStatement.getKey(), prepareStatement.getValue());
+            }
+            prepStmt.setInt(filterAttributeValue.entrySet().size() + 1, tenantId);
+            ResultSet rs = prepStmt.executeQuery();
+            if (rs.next()) {
+                return rs.getInt(1);
+            }
+        } catch (
+                SQLException e) {
+            throw new RuntimeException(e);
```

Comment:
```suggestion
        } catch (SQLException e) {
            throw new RuntimeException(e);
```

Reply:
Why are we throwing an runtime Exception here, This has to be properly handle

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13